src/overlays/misc/ovl_kaleido_scope/z_kaleido_collect.c

```diff
@@ -305,148 +325,161 @@ void KaleidoScope_DrawQuestStatus(PlayState* play, GraphicsContext* gfxCtx) {
-                    D_8082A0D8[sp218] -= sp226;
+            if (sMedEnvTimer != 0) {
+                stepRed = ABS(sMedEnvRed[j] - sMedEnvColors[targetColorIndex][0]) / sMedEnvTimer;
+                stepGreen = ABS(sMedEnvGreen[j] - sMedEnvColors[targetColorIndex][1]) / sMedEnvTimer;
+                stepBlue = ABS(sMedEnvBlue[j] - sMedEnvColors[targetColorIndex][2]) / sMedEnvTimer;
+                if (sMedEnvRed[j] >= sMedEnvColors[targetColorIndex][0]) {
+                    sMedEnvRed[j] -= stepRed;
                 } else {
-                    D_8082A0D8[sp218] += sp226;
+                    sMedEnvRed[j] += stepRed;
                 }
-                if (D_8082A0E4[sp218] >= D_8082A090[phi_v1][1]) {
-                    D_8082A0E4[sp218] -= sp224;
+                if (sMedEnvGreen[j] >= sMedEnvColors[targetColorIndex][1]) {
+                    sMedEnvGreen[j] -= stepGreen;
                 } else {
-                    D_8082A0E4[sp218] += sp224;
+                    sMedEnvGreen[j] += stepGreen;
                 }
-                if (D_8082A0F0[sp218] >= D_8082A090[phi_v1][2]) {
-                    D_8082A0F0[sp218] -= sp222;
+                if (sMedEnvBlue[j] >= sMedEnvColors[targetColorIndex][2]) {
+                    sMedEnvBlue[j] -= stepBlue;
                 } else {
-                    D_8082A0F0[sp218] += sp222;
+                    sMedEnvBlue[j] += stepBlue;
                 }
             } else {
-                D_8082A0D8[sp218] = D_8082A090[phi_v1][0];
-                D_8082A0E4[sp218] = D_8082A090[phi_v1][1];
-                D_8082A0F0[sp218] = D_8082A090[phi_v1][2];
+                sMedEnvRed[j] = sMedEnvColors[targetColorIndex][0];
+                sMedEnvGreen[j] = sMedEnvColors[targetColorIndex][1];
+                sMedEnvBlue[j] = sMedEnvColors[targetColorIndex][2];
             }
         }
 
-        if (CHECK_QUEST_ITEM(QUEST_MEDALLION_FOREST + sp218)) {
+        if (CHECK_QUEST_ITEM(QUEST_MEDALLION_FOREST + j)) {
             gDPPipeSync(POLY_OPA_DISP++);
             gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
-            gDPSetEnvColor(POLY_OPA_DISP++, D_8082A0D8[sp218], D_8082A0E4[sp218], D_8082A0F0[sp218], 0);
-            gSPVertex(POLY_OPA_DISP++, &pauseCtx->questVtx[sp21A], 4, 0);
+            gDPSetEnvColor(POLY_OPA_DISP++, sMedEnvRed[j], sMedEnvGreen[j], sMedEnvBlue[j], 0);
+            gSPVertex(POLY_OPA_DISP++, &pauseCtx->questVtx[bufI], 4, 0);
 
-            KaleidoScope_DrawQuadTextureRGBA32(gfxCtx, gItemIcons[ITEM_MEDALLION_FOREST + sp218], QUEST_ICON_WIDTH,
+            KaleidoScope_DrawQuadTextureRGBA32(gfxCtx, gItemIcons[ITEM_MEDALLION_FOREST + j], QUEST_ICON_WIDTH,
                                                QUEST_ICON_HEIGHT, 0);
         }
     }
 
-    if (D_8082A0FC == 0) {
-        D_8082A0FC = ZREG(61 + D_8082A100);
-        if (++D_8082A100 >= 4) {
-            D_8082A100 = 0;
+    if (sMedEnvTimer == 0) {
```

Comment:
I don't think "medallion" is abbreviated like this anywhere else at it could be confused for "medium", so I'd just spell out "medallion" everywhere